### ISSUE TITLE
[Dev branch] [Selfbot] OnMessageReceived DM messages content is empty

### ISSUE THREAD

**Issue by sezonis** [not-ilinked/Anarchy#3224]
`args.Message` is not null but `args.Message.Content` is always empty for DM's. Where Channel type is DM (untested with Group). 

`DiscordGatewayIntent.DirectMessages` is being passed through the config and this issue does not persist with the master branch, only the dev branch. Message `Id` is being returned correctly, but the problem is content is just empty no matter what. I'm not sure if this is recent, or I am doing something wrong with the new build. 

Messages are working perfectly fine for text channels though.

**Comment by not-ilinked** [comment]
I am unable to reproduce this. Can you share your code?

Here's mine for the record:
class Program
    {
        static void Main(string[] args)
        {
            DiscordSocketClient client = new DiscordSocketClient(new DiscordSocketConfig() { Intents = DiscordGatewayIntent.DirectMessages });
            client.OnLoggedIn += Client_OnLoggedIn;
            client.OnMessageReceived += Client_OnMessageReceived;
            client.Login("Bot ODUwNDA0NDExMTE2NzQ4ODIx.YLpO0w.CcVOS0NdLNeyYA_cVm6RCmURV0E");

            Thread.Sleep(-1);
        }

        private static void Client_OnMessageReceived(DiscordSocketClient client, MessageEventArgs args)
        {
            if (args.Message.Guild == null)
            {
                if (string.IsNullOrEmpty(args.Message.Content)) Console.WriteLine("No content");
                else Console.WriteLine(args.Message.Content);
            }
        }

        private static void Client_OnLoggedIn(DiscordSocketClient client, LoginEventArgs args)
        {
            Console.WriteLine("Logged in");
        }
    }

**Comment by sezonis** [comment]
The code is pretty much the same except for 2 things. 
1. The config
```
 static DiscordSocketClient client = new DiscordSocketClient(new DiscordSocketConfig() {
            Intents = DiscordGatewayIntent.DirectMessages | DiscordGatewayIntent.DirectMessageTyping | DiscordGatewayIntent.GuildMembers
            | DiscordGatewayIntent.GuildMessages | DiscordGatewayIntent.Guilds | DiscordGatewayIntent.GuildPresences | DiscordGatewayIntent.DirectMessageReactions
        });
        
            client.Login("user token here");
            client.OnLoggedIn += Client_OnLoggedIn; 
            client.OnMessageReceived += Client_OnMessageReceived;
        ```
2. the token I am using is a user token, not a bot token

But I did realize something strange. When I message the account that's listening to on message, it returns empty content. But when the account that's listening to on message messages others, the content is shown. So it's pretty much the content is 1 sided. 

To reproduce, you should be able to just message the selfbot account that's listening to OnMessage with any account, and the content will be empty. I pulled the latest changes from dev branch to ensure no bias. The result was the same, empty content.

**Comment by not-ilinked** [comment]
I tried the same stuff as i wrote above with a user token and it still worked without fail. No, Discord does not care about intents on user accounts.

**Comment by sezonis** [comment]
> I tried the same stuff as i wrote above with a user token and it still worked without fail. No, Discord does not care about intents on user accounts.

Could you give me the dll? I just want to make sure it isn't an issue on my end. I did recompile it a couple of times though. 

Try it with this dll (if you want)
[Anarchy.zip](https://github.com/not-ilinked/Anarchy/files/6644504/Anarchy.zip)

 and see if it still shows. I'm honestly confused because on the master branch it works really well

**Comment by not-ilinked** [comment]
I'm building it from the dev branch

**Comment by sezonis** [comment]
> I'm building it from the dev branch

After further debugging I believe I have figured out the culprit. This was due to supplying intents on the user token. 

If the token is a user token, supplying intents will make the message content blank. I had confused myself because I had 2 tokens running. One was a bot and one was a user token.  When I saw " ArgumentNullException("Gateway intents must be provided as of API v8");" I had assumed it to be for both tokens, as I didn't read the code of what was being thrown. But now after I looked at it, I saw this only applies to bot tokens.

**So what happened?** 

If you supply intents as a user, things start to bug out. It's as simple as that. Once I removed the custom intents, I could see the message contents!

**A possible fix to reduce confusion in the future:** Would be to ignore custom Intents that are passed in,  unless it is a bot token. 

I guess that solves this issue. The fix would probably help prevent this from happening in the future. 

If you want to recreate this, 

Just copy this directly and login with a user token 

```
        DiscordSocketClient client = new DiscordSocketClient(new DiscordSocketConfig() {
            Intents = DiscordGatewayIntent.DirectMessages,
        });
```        
Then messages will be blank when you message said user that's listening to OnMessage. I tested it on the "message listener" example project to ensure everything was as basic as it could be. 

So sorry if I may have caused some confusion for you.
